File: dev-docs/snippets/me/get.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Read the signed-in user's profile, using the cache when available.
export async function get(client: JsBaoClient) {
  // #region example
  const profile = await client.me.get({
    waitForLoad: "localIfAvailableElseNetwork",
    refreshIfOlderThanMs: 60_000,
  });
  // #endregion example
  return profile;
}

File: dev-docs/snippets/me/get.swift
import JsBaoClient

// Read the signed-in user's profile, using the cache when available.
func get(client: JsBaoClient) async throws {
  // #region example
  let profile = try await client.me.get(options: FetchCachedOptions(
    waitForLoad: .localIfAvailableElseNetwork,
    refreshIfOlderThanMs: 60_000
  ))
  // #endregion example
  _ = profile
}
